apps/version.c

```diff
@@ -31,6 +31,8 @@
 # include <openssl/blowfish.h>
 #endif
 
+#include "crypto/rand.h"
```

Comment:
I really really dislike this.

Reply:
What exactly do you dislike? I suppose it's the fact that a crypto-only header is used by the app? Would you prefer if I just reverted 7b18d1a53f932391bbc599a4717d6f98a597849c?

Reply:
> I suppose it's the fact that a crypto-only header is used by the app?

Yes.  Let's not confuse those boundaries.

> Would you prefer if I just reverted 7b18d1a?

Yes, I think that's the more viable solution.  It's still ugly, but at least doesn't *add on* more ugliness.

(another solution might be to add a `OSSL_DEVRANDOM` in opensslconf.h, just a simple definition without any ifdef sallad, and use that instead)

Reply:
> > Would you prefer if I just reverted [7b18d1a](https://github.com/openssl/openssl/commit/7b18d1a53f932391bbc599a4717d6f98a597849c)?
> 
> Yes, I think that's the more viable solution. It's still ugly, but at least doesn't _add on_ more ugliness.

The reverting commit was already prepared, just pushed it. See #10764.

Reply:
> (another solution might be to add a `OSSL_DEVRANDOM` in opensslconf.h, just a simple definition without any ifdef sallad, and use that instead)

Nope. That means adding a new feature to a public header.

Reply:
> Nope. That means adding a new feature to a public header.

...  there are times when we get overly strict with that...  but fine.

Reply:
> ... there are times when we get overly strict with that... but fine.

This regression is the example that we should be careful with adding unnecessary 'little' changes to stable branches. It wasn't even in public headers, but still broke some code.

> without any ifdef sallad

I agree with you that the entropy source ifdef salad is a technical debt and really needs to be cleaned up (in master).